lua/opencode/core.lua

```diff
@@ -51,6 +51,14 @@ function M.open(opts)
   local are_windows_closed = state.windows == nil
 
   if are_windows_closed then
+    -- Check if opening buffer is allowed
+    local allowed, err_msg = util.check_prompt_allowed(config.prompt_guard)
```

Comment:
Not sure I get why it's needed here ?

It would prevent from opening the plugin.